DigitalLearningSolutions.Web/ControllerHelpers/DateValidator.cs

```diff
@@ -1,172 +1,135 @@
-﻿namespace DigitalLearningSolutions.Web.ControllerHelpers
-{
-    using System;
-    using System.Collections.Generic;
-    using System.Linq;
-
-    public static class DateValidator
-    {
-        public class ValidationResult
-        {
-            public ValidationResult()
-            {
-                DateValid = true;
-                ErrorMessage = "";
-                DayValid = true;
-                MonthValid = true;
-                YearValid = true;
-            }
-
-            public ValidationResult(int day, int month, int year)
-            {
-                DateValid = true;
-                ErrorMessage = "";
-                DayValid = true;
-                MonthValid = true;
-                YearValid = true;
-                Day = day;
-                Month = month;
-                Year = year;
-            }
-
-            public bool DateValid { get; set; }
-            public string ErrorMessage { get; set; }
-            public bool DayValid { get; set; }
-            public bool MonthValid { get; set; }
-            public bool YearValid { get; set; }
-            public int? Day { get; }
-            public int? Month { get; }
-            public int? Year { get; }
-        }
-
-        public static ValidationResult ValidateRequiredDate(int? day, int? month, int? year, string name)
```

Comment:
It looks like this file was ported over into OldDateValidator, but OldDateValidator seems to be missing this method. Was it never used previously?

Reply:
I added this method for use in delegate registration by centre (parent ticket of this one), it didn't exist beforehand, so OldDateValidator doesn't need it. The combined functionality of OldDateValidator and this method that was added is all available in the ValidateDate method of the new DateValidator anyway since its refactored. But we need to keep OldDateValidator around until those LearningPortal pages gets refactored.